src/aws.ts

```diff
@@ -576,15 +578,21 @@ function makeDynamoIntegration<
       const table = getTableArgument(operationName, call.args);
       grantTablePermissions(table, context.role, operationName);
 
-      return {
-        Type: "Task",
-        Resource: `arn:aws:states:::aws-sdk:dynamodb:${operationName}`,
-        Parameters: context.toJson(
-          renameObjectProperties(input, {
-            Table: "TableName",
-          })
-        ),
-      };
+      const renamedExpr = renameObjectProperties(input, {
+        Table: "TableName",
+      });
+
+      return context.evalExpr(renamedExpr, (output) => {
+        return context.outputState(
+          ASLGraph.applyConstantOrVariableToTask(
+            {
+              Type: "Task",
+              Resource: `arn:aws:states:::aws-sdk:dynamodb:${operationName}`,
+            },
+            output
+          )
```

Comment:
It was simpler before! Before `eval` couldn't return multiple states, json path, and or constant values. Which meant that some values could not be passed to this integration, etc.

The callbacks (there is only one callback here) are an attempt at reducing the leakiness of the API (see [comment above)](https://github.com/functionless/functionless/pull/315#discussion_r917298393)).

If we wanted to break this out, it could be written as...

```ts
const exprState = context.eval(renamedExpr);
const output = ASLGraph.getAslStateOutput(exprState);

const tempHeap = context.newHeapVariable();

const task = {
    Type: "Task",
    Resource: `arn:aws:states:::aws-sdk:dynamodb:${operationName}`,
    ResultPath: tempHeap,
    ...ASLGraph.isJsonPath(output) ? { InputPath: output.jsonPath } : { Parameters: output.value }
};

const joined = ASLGraph.joinSubStates(expr, exprState, state);

return {
   ...joined,
   output: {
       jsonPath: tempHeap
   }
};
```

Updated the `apply*` names to `ASLGraph.taskWithInput` and `ASLGraph.passWithInput`.